lib/model/Model.js

```diff
@@ -526,7 +528,22 @@ class Model {
       knexOrTrx = null;
     }
 
-    return (knexOrTrx || this.knex()).transaction(cb);
+    return (knexOrTrx || this.knex()).transaction((trx) => {
+      trx.runAfterTransactionMethods = [];
```

Comment:
extending trx object to save runAfterTransactionMethods and afterTransationMethodsBatch helpers.